### PR TITLE
Migrate Trello Source Connector to Low Code

### LINKED ISSUE #28120: Trello

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Migrating Source Trello to Low-Code CDK resolves #28120

## How
Created the connector using the connector buider UI, then imported the manifest.yaml to the low code cdk.

## Recommended reading order
1. `manifest.yaml`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-trello/source_trello/components.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        If a parent slice contains no record, emit a slice with parent_record=None.
+
+        The template string can interpolate the following values:
+        - parent_stream_slice: mapping representing the parent's stream slice
+        - parent_record: mapping representing the parent record
+        - parent_stream_name: string representing the parent stream name
+        """
+        yielded_board_ids = set() # to track the board ids
+        config_board_ids = set(self.config.get("board_ids", [])) # board ids specified in config
+        for parent_stream_config in self.parent_stream_configs:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-trello/source_trello/components.py:34]: This function is huge! Is there anything we can do maybe to break it? Trying to think the real need of it :(

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-trello/source_trello/components.py (hunk 1 above)
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-trello/source_trello/components.py]: Please review now @marcosmarxm. I have simplified the function.

File: airbyte-integrations/connectors/source-trello/source_trello/components.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    def stream_slices(self) -> Iterable[StreamSlice]:
+
+        stream_map = {stream_config.stream.name:stream_config.stream for stream_config in self.parent_stream_configs}
+
+        board_ids = set(self.config.get("board_ids", []))
+
+        for board_id in read_all_boards(stream_boards=stream_map["boards"], stream_organizations=stream_map["organizations"]):
+            if not board_ids or board_id in board_ids:
+                yield {"id": board_id}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-trello/source_trello/components.py:41]: I think this is more aligned with what is expected using the `board_ids` parameter in the connector spec.
```suggestion
        board_ids = set(self.config.get("board_ids", []))
        if not board_ids:
            board_ids = read_all_boards(stream_boards=stream_map["boards"], stream_organizations=stream_map["organizations"])
        for board_id in board_ids:
            yield {"id": board_id}
```

This way you can change the `read_all_boards` to return a set/list of boards only when is needed.

**Review by marcosmarxm — DISMISSED** [review]

**Comment by topefolorunso** [comment]
@marcosmarxm Please review now

**Review by archangelic — CHANGES_REQUESTED** [review]
Can you add a changelog entry to `docs/integrations/sources/trello.md` and make sure the acceptance tests work after making the suggested changes?

File: airbyte-integrations/connectors/source-trello/unit_tests/test_order_ids_partition_router.py
[the remark this answers is not in the document]
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-trello/unit_tests/test_order_ids_partition_router.py:1]: @marcosmarxm requested the unit tests here https://github.com/airbytehq/airbyte/pull/29876#discussion_r1314920682

Do you still want me to remove it?

[the remark this answers is not in the document]
Comment (archangelic) [reply airbyte-integrations/connectors/source-trello/unit_tests/test_order_ids_partition_router.py:1]: I apologize for missing that. I've pushed some formatting changes to the tests (see [the docs](https://docs.airbyte.com/contributing-to-airbyte/resources/developing-locally/#run-formatting-automationtests)). However, they are currently failing on my end.

[the remark this answers is not in the document]
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-trello/unit_tests/test_order_ids_partition_router.py:1]: Can you share the error message? Maybe I can try to fix it

**Comment by topefolorunso** [comment]
Fixed the unit test, but I am unable to run the acceptance test on my end. Can you confirm on your end with your credentials? @archangelic

**Review by archangelic — APPROVED** [review]

**Comment by archangelic** [comment]
Everything looks good! Thanks for your contribution!

**Pull request merged**